src/app/visit-request/services/visit-request.service.ts

```diff
@@ -31,4 +32,20 @@ export class VisitRequestService {
   UpdateVisitRequestConfirmationById(visitRequestId: number): Observable<VisitRequestModel[]> {
     return this.httpClient.put<VisitRequestModel[]>(this.baseUrl + "VisitRequest/" + visitRequestId + "/Confirmation", this.httpOptions);
   }  
+
+    addVisitRequest(visitRequest: any) {
+      return this.httpClient.post<VisitRequestModel>(this.baseUrl + 'VisitRequest/', visitRequest, this.httpOptions)
+    }
+
+    getAllVisitRequestAvailability(visitRequestData: visitRequestAvailabilityDto) {
+      return this.httpClient.get(this.baseUrl + "VisitRequest/" + "Availability", {
```

Comment:
I would suggest to rename: getAllVisitRequestAvailability to getVisitRequestAvailability because we are just checking one visit request